CMakeLists.txt

```diff
@@ -33,8 +33,9 @@ endif()
 
 # interface target
 add_library(debug_assert INTERFACE)
-target_sources(debug_assert INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/debug_assert.hpp)
-target_include_directories(debug_assert INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+target_sources(debug_assert INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/debug_assert.hpp>)
+target_include_directories(debug_assert INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+target_include_directories(debug_assert SYSTEM INTERFACE $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>)
```

Comment:
This will add the includes with `-isystem` on compilers that support it, from gcc [documentation](https://gcc.gnu.org/onlinedocs/cpp/System-Headers.html):

> All warnings, other than those generated by ‘#warning’ (see Diagnostics), are suppressed while GCC is processing a system header.

You will still get warnings when building and testing the library, however, consumers of the library will not get warnings in your headers.